src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlException.cs

```diff
@@ -96,6 +96,28 @@ public override void GetObjectData(SerializationInfo si, StreamingContext contex
         /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlException.xml' path='docs/members[@name="SqlException"]/Class/*' />
         public byte Class => Errors.Count > 0 ? Errors[0].Class : default;
 
+#if NET6_0_OR_GREATER
+        /// <include file='../../../../../../doc/snippets/Microsoft.Data.SqlClient/SqlException.xml' path='docs/members[@name="IsTransient"]/IsTransient/*' />
+        public override bool IsTransient
+        {
+            get
+            {
+                if (Errors.Count == 0)
+                    return false;
+
+                for (int i = 0; i < Errors.Count; i++)
+                {
+                    if (!SqlConfigurableRetryFactory.IsKnownTransientErrorCode(Errors[i].Number))
```

Comment:
Thank you for this - it'll work for the general case. A user can also [configure SqlClient](https://learn.microsoft.com/en-us/sql/connect/ado-net/configurable-retry-logic-config-file-sqlclient?view=sql-server-ver16) to pass a custom set of error numbers to a built-in retry logic method, and can write their own SqlRetryLogicBase derivative with a custom TransientPredicate implementation (or a completely custom SqlRetryLogicBaseProvider derivative.) If this retry logic fails, it'll throw an AggregateException containing the various SqlException instances.

In a situation where the user's configured a custom list of transient errors (or defined their own TransientPredicate implementation), could they potentially receive an AggregateException containing a set of SqlExceptions where IsTransient is false?

If they could and this is a problem, one way to handle these custom sets of error numbers and the SqlRetryLogicBase derivative might be with some work in SqlRetryLogicProvider. I'm not sure how a SqlRetryLogicBaseProvider derivative could be handled though.